src/primitive_target.py

```diff
@@ -10,353 +9,48 @@
-    term: str
-    info: dict
-
-    @property
-    def value(self):
-        return json.dumps(self.info)
-
-
-def make_simple_transformation(type: str):
-    return {"transformation": {"type": type}}
-
-
-matching_transformation = ModifierTerm(
-    "matching", make_simple_transformation("matchingPairSymbol")
-)
-
-mod.list("cursorless_matching", desc="Supported symbol extent types")
-ctx.lists["self.cursorless_matching"] = {
-    matching_transformation.term: matching_transformation.value
-}
-
-containing_scope_type_map = {
-    "arg": "argumentOrParameter",
-    "arrow": "arrowFunction",
-    "call": "functionCall",
-    "class": "class",
-    "comment": "comment",
-    "element": "listElement",
-    "funk": "namedFunction",
-    "if": "ifStatement",
-    "key": "pairKey",
-    "lambda": "arrowFunction",
-    "list": "list",
-    "map": "dictionary",
-    "pair": "pair",
-    "state": "statement",
-    "string": "string",
-    "value": "value",
-}
-
-containing_scope_types = {
-    term: {
-        "transformation": {
-            "type": "containingScope",
-            "scopeType": containing_scope_type,
-        }
-    }
-    for term, containing_scope_type in containing_scope_type_map.items()
-}
-
-mod.list("containing_scope_type", desc="Supported symbol extent types")
-ctx.lists["self.containing_scope_type"] = {
-    key: json.dumps(value) for key, value in containing_scope_types.items()
-}
-
-SELECTION_TYPE_KEY = "selectionType"
-
-
-@dataclass
-class SelectionType:
-    singular: str
-    plural: str
-    json_name: str
-    rank: int
-
-    @property
-    def json_repr(self):
-        return {SELECTION_TYPE_KEY: self.json_name}
-
-
-TOKEN = SelectionType("token", "tokens", "token", 0)
-LINE = SelectionType("line", "lines", "line", 1)
-BLOCK = SelectionType("block", "blocks", "block", 2)
-FILE = SelectionType("file", "files", "document", 3)
-
-SELECTION_TYPES = [
-    TOKEN,
-    LINE,
-    BLOCK,
-    FILE,
-]
-
-ranked_selection_types = {
-    selection_type.json_name: selection_type.rank for selection_type in SELECTION_TYPES
-}
-
-cursor_mark = {"mark": {"type": "cursor"}}
-that_mark = {"mark": {"type": "that"}}
-
-marks = {
-    # "here": cursor_mark,  # Removed because conflicts with "air"
-    "this": cursor_mark,
-    **{
-        f"this {selection_type.singular}": {**selection_type.json_repr, **cursor_mark}
-        for selection_type in SELECTION_TYPES
-    },
-    **{
-        f"these {selection_type.plural}": {**selection_type.json_repr, **cursor_mark}
-        for selection_type in SELECTION_TYPES
-    },
-    "there": that_mark,
-    "that": that_mark,
-    "last cursor": {"mark": {"type": "lastCursorPosition"}},
-    **{
-        f"this {containing_scope_type}": {**cursor_mark, **value}
-        for containing_scope_type, value in containing_scope_types.items()
-    },
-    **{
-        f"that {containing_scope_type}": {**that_mark, **value}
-        for containing_scope_type, value in containing_scope_types.items()
```

Comment:
I believe we lose this ability, but prob not worth the complexity; I've never used it before 😄

Reply:
Yes just say `funk that` instead of `that funk`. Same with this. Simplified the grammar a bunch.